src/CentralNic/Helper/CentralNicApi.php

```diff
@@ -0,0 +1,507 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Upmind\ProvisionProviders\DomainNames\CentralNic\Helper;
+
+use Carbon\Carbon;
+use GuzzleHttp\Client;
+use Illuminate\Support\Arr;
+use Upmind\ProvisionBase\Helper;
+use Metaregistrar\EPP\eppCheckDomainRequest;
+use Metaregistrar\EPP\eppCheckDomainResponse;
+use Metaregistrar\EPP\eppCheckRequest;
+use Metaregistrar\EPP\eppPollResponse;
+use Metaregistrar\EPP\eppPollRequest;
+use Metaregistrar\EPP\eppContact;
+use Metaregistrar\EPP\eppContactHandle;
+use Metaregistrar\EPP\eppContactPostalInfo;
+use Metaregistrar\EPP\eppCreateContactRequest;
+use Metaregistrar\EPP\eppCreateContactResponse;
+use Metaregistrar\EPP\eppCreateDomainRequest;
+use Metaregistrar\EPP\eppCreateDomainResponse;
+use Metaregistrar\EPP\eppCreateHostRequest;
+use Metaregistrar\EPP\eppCreateResponse;
+use Metaregistrar\EPP\eppDomain;
+use Metaregistrar\EPP\eppException;
+use Metaregistrar\EPP\eppHost;
+use Metaregistrar\EPP\eppInfoContactRequest;
+use Metaregistrar\EPP\eppInfoContactResponse;
+use Metaregistrar\EPP\eppInfoDomainRequest;
+use Metaregistrar\EPP\eppInfoDomainResponse;
+use Metaregistrar\EPP\eppRenewRequest;
+use Metaregistrar\EPP\eppRenewResponse;
+use Metaregistrar\EPP\eppTransferRequest;
+use Metaregistrar\EPP\eppTransferResponse;
+use Metaregistrar\EPP\eppUpdateContactRequest;
+use Metaregistrar\EPP\eppUpdateContactResponse;
+use Metaregistrar\EPP\eppUpdateDomainRequest;
+use Metaregistrar\EPP\eppUpdateDomainResponse;
+use Metaregistrar\EPP\eppUpdateResponse;
+use Metaregistrar\EPP\eppResponse;
+use Upmind\ProvisionBase\Exception\ProvisionFunctionError;
+use Upmind\ProvisionBase\Provider\DataSet\SystemInfo;
+use Upmind\ProvisionProviders\DomainNames\CentralNic\EppExtension\EppConnection;
+use Upmind\ProvisionProviders\DomainNames\Data\ContactParams;
+use Upmind\ProvisionProviders\DomainNames\Data\DacDomain;
+use Upmind\ProvisionProviders\DomainNames\Data\DomainContactInfo;
+use Upmind\ProvisionProviders\DomainNames\Data\DomainNotification;
+use Upmind\ProvisionProviders\DomainNames\Helper\Utils;
+use Upmind\ProvisionProviders\DomainNames\CentralNic\Data\Configuration;
+
+/**
+ * Class CentralNicHelper
+ *
+ * @package Upmind\ProvisionProviders\DomainNames\CentralNic\Helper
+ */
+class CentralNicApi
+{
+    protected EppConnection $connection;
+    protected Configuration $configuration;
+
+    protected const CONTACT_LOC = 'loc';
+    protected const CONTACT_INT = 'int';
+    protected const CONTACT_AUTO = 'auto';
+
+    protected array $lockedStatuses = [
+        'clientTransferProhibited',
+        'clientUpdateProhibited',
+        'clientDeleteProhibited',
```

Comment:
I think setting this status may cause unintended effects with certain Registries (like preventing a domain from expiring when it's supposed to). Could you change this to only `clientTransferProhibited` and `clientUpdateProhibited`?